### PR TITLE
Document Asynchronous Configurations in the Config File's Docs

### PR DESCRIPTION
**Pull request by npetruzzelli**
In the process of trying to fix my branch, it looks like resetting it closed my original PR: #3733

Which was intended to support: https://github.com/karma-runner/karma/issues/3677#issuecomment-997002670

@jginsburgn @devoto13

### CODE REVIEW

File: docs/config/01-configuration-file.md [hunk 1]
```diff
@@ -67,6 +67,45 @@ require('ts-node').register({
...
+        frameworks: ['jasmine'],
+        port: freePort,
+        // ... other Karma configuration
+      });
+    });
+}
+```
+
+If you are using the `parseConfig` method from the public API, then please see
+the [Public API Documentation][dev/public-api]] for details, usage, and
```
Comment (jginsburgn) [inline docs/config/01-configuration-file.md:106]: Extra bracket :)

**Review by jginsburgn — CHANGES_REQUESTED** [review]

File: docs/config/01-configuration-file.md (hunk 1 above)
Comment (npetruzzelli) [reply docs/config/01-configuration-file.md:106]: That feeling when you  have checked something 999 times, but didn't go for 1,000.

I have this feeling now! I'll fix it up soon.

Comment (npetruzzelli) [reply docs/config/01-configuration-file.md]: @jginsburgn - I've fixed the typo and refined the example based on @nicojs 's [feedback in the original PR](https://github.com/karma-runner/karma/pull/3733#pullrequestreview-879956701)

**Comment by jginsburgn** [comment]
Sorry for the delay!

Please just rebase and let's get this PR in :)

**Comment by devoto13** [comment]
I don't know, I'm keener towards a simpler approach displaying the `async` function as in https://github.com/karma-runner/karma/pull/3761 because that's what people would use normally instead of noisy Promise chains.

**Comment by jginsburgn** [comment]
> I don't know, I'm keener towards a simpler approach displaying the `async` function as in #3761 because that's what people would use normally instead of noisy Promise chains.

You're right. I had forgotten. Is that ok with you @npetruzzelli ?

**Comment by jginsburgn** [comment]
@npetruzzelli I will close this tentatively. If you have objections. Please feel free to reopen.

**Pull request closed without merging**